eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/serialization/commons/Extra.java

```diff
@@ -0,0 +1,32 @@
+package com.schedjoules.eventdiscovery.framework.serialization.commons;
+
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.schedjoules.eventdiscovery.framework.serialization.core.Key;
+import com.schedjoules.eventdiscovery.framework.utils.factory.Factory;
+import com.schedjoules.eventdiscovery.framework.utils.optionals.AbstractCachingOptional;
+
+import org.dmfs.optional.Optional;
+
+
+/**
+ * @author Gabor Keszthelyi
+ */
+public final class Extra<T> extends AbstractCachingOptional<T>
+{
+    // TODO Implement Optional here, not use abstract class
+
+
+    public Extra(final Key<T> key, final Intent intent)
```

Comment:
Maybe consider creating a constructor that takes an activity or even context instead of an intent.